src/satisfia/util/helper.py

```diff
@@ -1,56 +1,42 @@
 #!/usr/bin/env python3
 
-class Interval():
-	def __init__(self, left, right=None):
-		if hasattr(left, "__len__"):
-			if right != None:
-				raise TypeError()
-			if len(left) != 2:
-				raise TypeError()
-			self._left = left[0]
-			self._right = left[1]
-		else:
-			self._left = left
-			self._right = left if (right == None) else right
+from typing import NamedTuple
+
+def Interval(left, right=None):
+    if hasattr(left, "__len__"):
+        assert right is None
+        return _Interval(*left)
+    else:
+        return _Interval(left,  left if right is None else right)
+
+class _Interval(NamedTuple):
```

Comment:
namedtuples are much faster than generic classes.

This change should be transparent to the rest of the code

Reply:
very good!